### PR TITLE
vferreira/feat/TDP-5322-tdatapreprun-spark-mode-test

### PR DESCRIPTION
**Pull request by VincentFerreira**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-5322

**Please check if the PR fulfills these requirements**
- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed on local, INT, tal-rd44 and many times
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/SparkComparator.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+// ============================================================================
```
[the remark this answers is not in the document]
Comment (yacarof) [reply dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/SparkComparator.java]: done

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/SparkComparator.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            throws IOException {
+        LineIterator actualLI = IOUtils.lineIterator(actual, StandardCharsets.UTF_8);
+        LineIterator expectedLI = IOUtils.lineIterator(expected, StandardCharsets.UTF_8);
+        Map<String, Integer> inActual = new HashMap<>();
+        Map<String, Integer> inExpected = new HashMap<>();
+        String actualLine;
+        String expectedLine;
+
+        while (actualLI.hasNext() || expectedLI.hasNext()) {
+            actualLine = actualLI.hasNext() ? actualLI.nextLine() : null;
```
Comment (elogeart) [inline dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/SparkComparator.java]: You could declare and instantiate `actualLine`  and `expectedLine` at the same time in this while loop.

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/FileStep.java [hunk 3]
```diff
@@ -64,4 +64,18 @@ public void thenICheckThatTheFileIsTheExpectedOne(String temporaryFilename, Stri
             }
         }
     }
+
+    @Then("^I check that \"(.*)\" temporary file contains the same lines than \"(.*)\" file$")
```
[the remark this answers is not in the document]
Comment (yacarof) [reply dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/FileStep.java]: done

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/FileStep.java [hunk 4]
```diff
@@ -64,4 +64,18 @@ public void thenICheckThatTheFileIsTheExpectedOne(String temporaryFilename, Stri
...
+    public void thenICheckThatTheSparkFileIsTheExpectedOne(String temporaryFilename, String expectedFilename)
```
[the remark this answers is not in the document]
Comment (yacarof) [reply dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/FileStep.java]: done

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/SparkComparator.java [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+import org.apache.commons.io.LineIterator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SparkComparator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SparkComparator.class);
+
+    /**
+     * Compare to file in order to check if they contains the same lines (but maybe not in the same order).
```
[the remark this answers is not in the document]
Comment (yacarof) [reply dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/SparkComparator.java]: done

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/SparkComparator.java [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+     * 
+     * @param actual the actual file
+     * @param expected the file expected to match the actual one
+     * @return <code>true</code> if both files contains the same lines, <code>false</code> else.
+     * @throws IOException in case of reading file problem.
+     */
+    public static boolean doesFilesContainSameLines(@NotNull InputStream actual, @NotNull InputStream expected)
```
[the remark this answers is not in the document]
Comment (yacarof) [reply dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/SparkComparator.java]: done

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/SparkComparator.java [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+            throws IOException {
+        LineIterator actualLI = IOUtils.lineIterator(actual, StandardCharsets.UTF_8);
+        LineIterator expectedLI = IOUtils.lineIterator(expected, StandardCharsets.UTF_8);
+        Map<String, Integer> inActual = new HashMap<>();
```
[the remark this answers is not in the document]
Comment (yacarof) [reply dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/SparkComparator.java]: done

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/SparkComparator.java (hunk 2 above)
Comment (yacarof) [reply dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/SparkComparator.java]: done

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/SparkComparator.java [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+import java.util.Map;
+
+import javax.validation.constraints.NotNull;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.LineIterator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SparkComparator {
```
Comment (elogeart) [inline dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/SparkComparator.java]: Please, remove mentions to Spark as this class is a file comparator, not related to Spark.

**Pull request merged**